src/app/core/service/auth.service.ts

```diff
@@ -0,0 +1,31 @@
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { map, Observable } from 'rxjs';
+import { LoginForm, RegisterForm, UserToken } from '../model';
+import { JwtHelperService } from '@auth0/angular-jwt';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+
+  private headers = new HttpHeaders({ 'Content-Type': 'application/json' });
+  private readonly path = 'auth';
+  private readonly helper = new JwtHelperService();
+
+  constructor(private http: HttpClient) { }
+
+  login(loginForm: LoginForm): Observable<void> {
+    return this.http.post<UserToken>(`${this.path}/login`, loginForm, { headers: this.headers })
```

Comment:
Idea to improve this to reduce code repeat:
Create new HttpClient which extends HttpClient with proper get and post methods. Inject headers and path in constructor.